### PR TITLE
support for predefined sample weights: ' balance', 'auto' options.

### PR DESCRIPTION
**Pull request by jwmueller**
extra string options for `sample_weight` argument.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-962-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-962/3/index.html

**Review by Innixma — APPROVED** [review]
Looks good, just minor comments

File: tabular/src/autogluon/tabular/learner/default_learner.py [hunk 1]
```diff
@@ -203,8 +209,23 @@ def general_data_processing(self, X: DataFrame, X_val: DataFrame, X_unlabeled: D
...
+            raise ValueError(f"Column name '{self.sample_weight}' cannot appear in your dataset with predefined weighting strategy. Please change it and try again.")
+        if self.sample_weight == BALANCE_WEIGHT:
+            if self.class_weights is None:
+                class_counts = y.value_counts()
+                self.class_weights = {c : len(y)/(class_counts[c]*len(class_counts)) for c in class_counts.index}
+                logger.log(15, f"Balancing classes via the following weights: {self.class_weights}")
+            w = y.map(self.class_weights)
+        elif self.sample_weight == AUTO_WEIGHT:  # TODO: support more sophisticated auto_weight strategy
+            raise NotImplementedError(f"{AUTO_WEIGHT} strategy not yet supported.")
+        X[self.sample_weight] = w
```
[the remark this answers is not in the document]
Comment (jwmueller) [reply tabular/src/autogluon/tabular/learner/default_learner.py:227]: added TODO: consider not bundling sample weights inside X

**Comment by szha** [comment]
Job PR-962-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-962/4/index.html

**Pull request merged**